terraform_module/main.tf

```diff
@@ -1,17 +1,6 @@
-resource "null_resource" "example1" {
+resource "null_resource" "examples" {
+  count = length(var.commands)
   provisioner "local-exec" {
-    command = "echo 'Executed example1'"
-  }
-}
-
-resource "null_resource" "example2" {
-  provisioner "local-exec" {
-    command = "echo 'Executed example2'"
-  }
-}
-
-resource "null_resource" "example3" {
-  provisioner "local-exec" {
-    command = "echo 'Executed example3'"
+    command = var.commands[count.index]
   }
 }
```

Comment:
This solution works as the example commands are only unique because they differ by sequential integers, but the first solution you tried as more general. Changing the input variable type or using one of terraform's type conversion functions would have made the first solution work.